libcxx/test/libcxx/ranges/range.adaptors/range.adaptor.helpers/as-lvalue.cpp

```diff
@@ -0,0 +1,47 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
+
+// <ranges>
+
+// template<class T>
+// constexpr T& as-lvalue(T&& t) { // exposition only
+
+#include <ranges>
+
+#include <concepts>
+#include <utility>
+
+constexpr bool test() {
+  // Check glvalue
+  {
+    int lvalue{};
+    [[maybe_unused]] std::same_as<int&> decltype(auto) check = std::ranges::__as_lvalue(lvalue);
+  }
+
+  // Check prvalue (reference is dangling)
+  {
+    [[maybe_unused]] std::same_as<int&> decltype(auto) check = std::ranges::__as_lvalue(0);
```

Comment:
Yeah, this test show exactly how dangerous this API is.

Why do you need it?

Reply:
The *`as-lvalue`* function is used by P2770 to fix joining ranges of xvalues (rvalue references), see [LWG-3791](https://cplusplus.github.io/LWG/issue3791) and example: https://godbolt.org/z/Ev17W3b5W.